homeassistant/components/sensor/snmp.py

```diff
@@ -105,19 +121,30 @@ def unit_of_measurement(self):
     def update(self):
         """Get the latest data and updates the states."""
         self.data.update()
-        self._state = self.data.value
+        value = self.data.value
+
+        if value is None:
+            value = STATE_UNKNOWN
+        elif self._value_template is not None:
+            value = self._value_template.render_with_possible_json_value(
```

Comment:
It's possible and entirely device-dependent, as SNMP can return string values. Any manufacturer can decide that one of their OIDs is going to return a JSON representation of some data. I'd rather leave the opportunity and cover the odd edge case.